lib/Byng/Pimcore/Sitemap/Generator/SitemapGenerator.php

```diff
@@ -65,18 +66,51 @@ public function __construct()
      */
     public function generateXml()
     {
-        // Get all the root elements with parentId '1'
-        $rootDocuments = $this->documentGateway->getChildren(1);
-
-        foreach ($rootDocuments as $rootDocument) {
-            $this->addUrlChild($rootDocument);
-            $this->listAllChildren($rootDocument);
+        // Retrieve site trees
+        $siteRoots = new Site\Listing();
+        $siteRoots = $siteRoots->load();
+
+        // Build siteRoots table: [ ID => Domain ]
+        /* @var Site $siteRoot */
+        foreach ($siteRoots as $siteRoot){
+            $this->sitesRoots[$siteRoot->getRootId()] = $siteRoot->getMainDomain();
         }
-        $this->xml->asXML(PIMCORE_DOCUMENT_ROOT . "/sitemap.xml");
 
-        if (Config::getSystemConfig()->get("general")->get("environment") === "production") {
-            $this->notifySearchEngines();
+        // Also append the default tree
+        $this->sitesRoots[1] = Config::getSystemConfig()->get("general")->get("domain");
+
+        $notifySearchEngines = Config::getSystemConfig()->get("general")->get("environment") === "production";
+        foreach($this->sitesRoots as $siteRootID => $siteRootDomain){
+            $this->generateSiteXml($siteRootID, $siteRootDomain);
+
+            if ($notifySearchEngines)
+                $this->notifySearchEngines('https://' . $siteRootDomain);
         }
+
+    }
+
+    /**
+     * Generate a sitemap xml file for a defined site, with a specific hostUrl
+     *
+     * @param $rootId
+     * @param $hostUrl
+     * @return string
+     */
+    private function generateSiteXml($rootId, $hostUrl)
+    {
+        $this->xml = new SimpleXMLElement(
+            '<?xml version="1.0" encoding="UTF-8"?>'
+            . '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"></urlset>'
+        );
+
+        // Set current hostUrl
+        $this->hostUrl = 'https://' . $hostUrl;
```

Comment:
Hey, as I explained in the summary, I found no way to get the protocol of the website in Pimcore settings 😕 

I thought of the following possible resolutions, but I'm not quite happy with any of them: 
- Creating a new variable in `Settings > Website` tab (assumes all the sites and subsites share the same protocol)
- create custom plugin config file to set parameters for each site?
- try curl request to https version of the hosturl, use http if error is returned (then cache the outcome)

If you have any idea...